lib/ansible/modules/packaging/os/yum.py

```diff
@@ -390,6 +391,44 @@ def __init__(self, module):
         self.pkg_mgr_name = "yum"
         self.lockfile = '/var/run/yum.pid'
 
+    def is_lockfile_pid_valid(self):
+        try:
+            with open(self.lockfile, 'r') as f:
+                oldpid = int(f.readline())
+        except ValueError:
+            # invalid data
+            os.unlink(self.lockfile)
+            return False
+        except (IOError, OSError) as e:
+            self.module.fail_json(msg="Failure opening %s: %s" % (self.lockfile, to_native(e)))
```

Comment:
If the lockfile has gone away while you were attempting to read it, wouldn't you want the ansible module to continue instead of failing?

Related, if the lockfile disappears at any time while we're attempting to decide if it's valid, we likely want the module to continue.  So we probably want to ```try-except``` around the body of this function and return False if the exception tells us that the lockfile no longer exists.

Reply:
:+1: